runtime/Python3/src/antlr4/xpath/XPath.py

```diff
@@ -246,18 +254,16 @@ def evaluate(self, t:ParseTree):
         dummyRoot.children = [t] # don't set t's parent.
 
         work = [dummyRoot]
-
-        for i in range(0, len(self.elements)):
-            next = set()
+        for element in self.elements:
+            work_next = list()
             for node in work:
-                if len( node.children) > 0 :
+                if not isinstance(node, TerminalNode) and node.children:
```

Comment:
That is probably wrong,g, not sure there is guarantee that node.children is never empty

Reply:
https://www.python.org/dev/peps/pep-0008/
>For sequences, (strings, lists, tuples), use the fact that empty sequences are false.
>
>```Python3
>Yes: if not seq:
>     if seq:
>
>No:  if len(seq):
>     if not len(seq):
>```

Reply:
Not sure I support that pep...